src/main/java/org/jmt/mcmt/serdes/pools/SingleExecutionPool.java

```diff
@@ -2,24 +2,29 @@
 
 import java.util.concurrent.locks.Lock;
 import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Consumer;
 
 import javax.annotation.Nullable;
 
 import net.minecraft.util.math.BlockPos;
 import net.minecraft.world.World;
 
+/**
+ * Run on main thread.
+ */
 public class SingleExecutionPool implements ISerDesPool {
 
 	private Lock l = new ReentrantLock();
-	
+
 	@Override
-	public void serialise(Runnable task, Object o, BlockPos bp, World w, @Nullable ISerDesOptions options) {
+	public void serialise(Runnable task, Object o, BlockPos bp, World w,
+			Consumer<Runnable> executeMultithreaded, @Nullable ISerDesOptions options) {
 		try {
```

Comment:
This should be being pushed to the executeMultithreaded as it's allowed to execute off main thread, just not more than one at a time

The idea for this pool was global stuff like enderchests/tanks or equiv where you have a globally accessible single data element that will probably break